gef-ui/server/controllers/unissued-facilities/facility-end-date/index.ts

```diff
@@ -0,0 +1,217 @@
+import { Response } from 'express';
+import { getDate, getMonth, getYear, isSameDay, parseISO, startOfDay } from 'date-fns';
+import { CustomExpressRequest, isFacilityEndDateEnabledOnGefVersion, parseDealVersion } from '@ukef/dtfs2-common';
+import { isTrueSet, validationErrorHandler } from '../../../utils/helpers';
+import * as api from '../../../services/api';
+import { validateAndParseFacilityEndDate } from './validation';
+import { asLoggedInUserSession } from '../../../utils/express-session';
+import { FacilityEndDateViewModel } from '../../../types/view-models/facility-end-date-view-model';
+import { ValidationError } from '../../../types/validation-error';
+
+type FacilityEndDateParams = { dealId: string; facilityId: string };
+type FacilityEndDatePostBody = { 'facility-end-date-day': string; 'facility-end-date-month': string; 'facility-end-date-year': string };
+
+type GetFacilityEndDateRequest = CustomExpressRequest<{ params: FacilityEndDateParams; query: { change: string } }>;
+type PostFacilityEndDateRequest = CustomExpressRequest<{
+  reqBody: FacilityEndDatePostBody;
+  params: FacilityEndDateParams;
+  query: { saveAndReturn: string };
+}>;
+
+const getFacilityEndDateViewModel = async (req: GetFacilityEndDateRequest, previousPage: string): Promise<FacilityEndDateViewModel | null> => {
+  const {
+    params: { dealId, facilityId },
+    session: { userToken },
+  } = req;
+
+  const { details: facility } = (await api.getFacility({ facilityId, userToken })) as { details: Record<string, unknown> };
+  const deal = (await api.getApplication({ dealId, userToken })) as Record<string, unknown> & { version?: number };
+
+  if (!isFacilityEndDateEnabledOnGefVersion(parseDealVersion(deal.version)) || !facility.isUsingFacilityEndDate) {
+    return null;
+  }
+
+  const facilityEndDateViewModel: FacilityEndDateViewModel = {
+    dealId,
+    facilityId,
+    previousPage,
+  };
+
+  if ('facilityEndDate' in facility && typeof facility.facilityEndDate === 'string') {
```

Comment:
Does `if (typeof facility?.facilityEndDate === 'string')` perform the same?

Reply:
I think the `?` isnt even needed - was being overly cautious but `facility` is definitely defined